cmake/os_flags.cmake

```diff
@@ -126,6 +126,28 @@ function(ie_avx512_optimization_flags flags)
     endif()
 endfunction()
 
+function(ie_arm_neon_optimization_flags flags)
```

Comment:
Why do we need to move this method to common part?

Is it needed to use this method in external modules?

Reply:
yes, ARM plugin.
Secondly, it's a common functionality which allows to get proper NEON flags.